src/main/java/io/codelex/flightplanner/repository/FlightPlannerRepository.java

```diff
@@ -0,0 +1,122 @@
+        if (isExistingFlight(id)) {
+            return findExistingFlight(id);
+        }
+        throw new ResponseStatusException(HttpStatus.NOT_FOUND);
+    }
+
+    public synchronized void deleteFlight(int id) {
+        if (isExistingFlight(id)) {
+            Flight foundFlight = findExistingFlight(id);
+            flights.remove(foundFlight);
+        }
+    }
+
+    public List<Airport> searchAirports(String search) {
+        List<Airport> airportList = getExistingAirportList(search);
+        if (airportList.isEmpty()) {
+            throw new ResponseStatusException(HttpStatus.NOT_FOUND);
+        }
+        return airportList;
+    }
+
+    public PageResult searchFlights(SearchFlightRequest flight) {
+        PageResult pageResult = getExistingFlight(flight);
+        if (flight.getTo().equals(flight.getFrom())) {
+            throw new ResponseStatusException(HttpStatus.BAD_REQUEST);
+        }
+        return pageResult;
+    }
+
+    private boolean isFromAndToSameAirport(Flight flight) {
+        String fromAirportTrimmed = flight.getFrom().getAirport().replace(" ", "");
+        String toAirportTrimmed = flight.getTo().getAirport().replace(" ", "");
+        return fromAirportTrimmed.equalsIgnoreCase(toAirportTrimmed);
+    }
+
+    private boolean hasInvalidDates(Flight flight) {
+        return flight.getDepartureTime().isEqual(flight.getArrivalTime())
+                || flight.getDepartureTime().isAfter(flight.getArrivalTime());
+    }
+
+    private boolean isExistingFlight(int id) {
+        return flights.stream()
+                .map(Flight::getId)
+                .anyMatch(integer -> integer == id);
+    }
+
+    private Flight findExistingFlight(int id) {
+        return FlightPlannerRepository.this.flights.stream()
+                .filter(flight -> flight.getId() == id)
+                .findFirst()
+                .orElse(null);
+    }
+
+    private List<Airport> getExistingAirportList(String search) {
+        List<Airport> airportList = new ArrayList<>();
+        String searchTrimmedLowerCase = search.replace(" ", "").toLowerCase();
+        for (Flight flight : flights) {
+            if (flight.getFrom().getCountry().toLowerCase().contains(searchTrimmedLowerCase)
+                    || flight.getFrom().getAirport().toLowerCase().contains(searchTrimmedLowerCase)
+                    || flight.getFrom().getCity().toLowerCase().contains(searchTrimmedLowerCase)) {
+                airportList.add(flight.getFrom());
+            }
+        }
+        return airportList;
+    }
+
+    private PageResult getExistingFlight(SearchFlightRequest flight) {
+        PageResult pageResult = new PageResult();
+        for (Flight value : flights) {
+            if (value.getFrom().getAirport().equals(flight.getFrom())
+                    && value.getTo().getAirport().equals(flight.getTo())
+                    && String.valueOf(value.getDepartureTime()).contains(flight.getDepartureDate())) {
```

Comment:
Te nevajadzētu konvertēt uz String, var izmantot equals.
Man liekas kaut kā šādi aptuveni:
`value.getDepartureTime().toLocalDate().equals( .. )`

Reply:
Visu izlaboju izņemot šo, lieta tāda, ka `SearchFlightRequest` klase glabā `departureDate` kā String un viņš tiek padots kā vienkārši datums bez laika (piemēram, 2019-01-01), bet `value`, kas ir `Flight` objekts - viņam tiek glabāts `departureTime` un viņš tiek glabāts kā LocalDateTime ar laiku (piemēram, 2019-01-01 10:31).
Es nevaru salīdzināt ar .equals, jo viens ir LocalDateTime objekts, otrs ir String. Varu parsēt abus mainīgos (`departureDate` un `departureTime`) kā LocalDate un tad salīdzināt ar .equals, tas būtu labāks variants par šo String .contains variantu?

Reply:
Tad nomaini SearchFlightRequest klasē departureDate uz LocalDate. Man liekas tam vajadzētu strādāt out-of-the-box. Viņam bija problēmas konvertēt no JSON uz LocalDateTime, tāpēc, ka tas tests tur citā formātā sūtīja tos datus, bet uz LocalDate nevajadzētu būt problēmām. Tas arī uzreiz atrisinās šo. :)